web/src/lib/components/Navbar.svelte

```diff
@@ -72,64 +86,99 @@
-                    <li><a href={null} on:click|preventDefault={() => { token.set(null); localStorage.removeItem('token'); goto("/"); }} class="modal-button cursor-pointer">Logout</a></li>
-                </ul>
+    <label class="rounded btn btn-ghost swap swap-rotate" on:click={toggleTheme} on:click={closeMenu}>
+        <input id="theme-toggle" type="checkbox" checked={prefersDark} />
+        <!-- sun icon -->
+        <svg class="swap-off fill-current w-10 h-10" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M5.64,17l-.71.71a1,1,0,0,0,0,1.41,1,1,0,0,0,1.41,0l.71-.71A1,1,0,0,0,5.64,17ZM5,12a1,1,0,0,0-1-1H3a1,1,0,0,0,0,2H4A1,1,0,0,0,5,12Zm7-7a1,1,0,0,0,1-1V3a1,1,0,0,0-2,0V4A1,1,0,0,0,12,5ZM5.64,7.05a1,1,0,0,0,.7.29,1,1,0,0,0,.71-.29,1,1,0,0,0,0-1.41l-.71-.71A1,1,0,0,0,4.93,6.34Zm12,.29a1,1,0,0,0,.7-.29l.71-.71a1,1,0,1,0-1.41-1.41L17,5.64a1,1,0,0,0,0,1.41A1,1,0,0,0,17.66,7.34ZM21,11H20a1,1,0,0,0,0,2h1a1,1,0,0,0,0-2Zm-9,8a1,1,0,0,0-1,1v1a1,1,0,0,0,2,0V20A1,1,0,0,0,12,19ZM18.36,17A1,1,0,0,0,17,18.36l.71.71a1,1,0,0,0,1.41,0,1,1,0,0,0,0-1.41ZM12,6.5A5.5,5.5,0,1,0,17.5,12,5.51,5.51,0,0,0,12,6.5Zm0,9A3.5,3.5,0,1,1,15.5,12,3.5,3.5,0,0,1,12,15.5Z"/></svg>
+        <!-- moon icon -->
+        <svg class="swap-on fill-current w-10 h-10" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M21.64,13a1,1,0,0,0-1.05-.14,8.05,8.05,0,0,1-3.37.73A8.15,8.15,0,0,1,9.08,5.49a8.59,8.59,0,0,1,.25-2A1,1,0,0,0,8,2.36,10.14,10.14,0,1,0,22,14.05,1,1,0,0,0,21.64,13Zm-9.5,6.69A8.14,8.14,0,0,1,7.08,5.22v.27A10.15,10.15,0,0,0,17.22,15.63a9.79,9.79,0,0,0,2.1-.22A8.11,8.11,0,0,1,12.14,19.73Z"/></svg>
+    </label>
+    <div class="ml-auto pb-2 pt-1 pr-1" on:click={toggleMenu}>
+        {#if $token && $user && $user.twitterProfileImageUrl}
+        <label class="btn btn-ghost btn-circle avatar ring-2 ring-primary ring-offset-0" for={null} class:verified={$user.twitterUsernameVerified} class:not-verified={!$user.twitterUsernameVerified}>
+            <div class="w-10 rounded-full">
+                <img class="overflow-hidden" src={getProfileImage($user)} alt="Avatar" />
             </div>
+        </label>
+        {:else}
+        <label class="btn btn-ghost btn-circle avatar ring-2 ring-primary ring-offset-0" for={null} >
+            <svg xmlns="http://www.w3.org/2000/svg" width="36" height="36" fill="currentColor" class="bi bi-list" viewBox="0 0 16 16">
+                <path fill-rule="evenodd" d="M2.5 12a.5.5 0 0 1 .5-.5h10a.5.5 0 0 1 0 1H3a.5.5 0 0 1-.5-.5zm0-4a.5.5 0 0 1 .5-.5h10a.5.5 0 0 1 0 1H3a.5.5 0 0 1-.5-.5zm0-4a.5.5 0 0 1 .5-.5h10a.5.5 0 0 1 0 1H3a.5.5 0 0 1-.5-.5z"/>
+            </svg>
+        </label>
         {/if}
     </div>
-</div>
+    <div class="{menuDisplay} w-full">
+        <ul class="flex flex flex-col" on:click={closeMenu}>
+            <a class="md:hidden px-3 py-2 rounded btn btn-ghost" href="{getBaseUrl()}about">
```

Comment:
Why do we need to use getbaseurl() here? Any reason why we don't simply link to /about?